provider/kubernetes/kubernetes.go

```diff
@@ -421,6 +443,27 @@ func getRuleForPath(pa extensionsv1beta1.HTTPIngressPath, i *extensionsv1beta1.I
 		}
 		rules = append(rules, ruleTypeReplacePath+":"+rootPath)
 	}
+
+	if requestModifier := getStringValue(i.Annotations, annotationKubernetesRequestModifier, ""); requestModifier != "" {
```

Comment:
What happens if the annotation consists of all whitespaces only?

Reply:
Rude

Reply:
Also, added tests to include a bunch of other oddities